Makefile

```diff
@@ -59,49 +131,63 @@ dev/%: ## run a foreground container for a stack
 dev-env: ## install libraries required to build docs and run tests
 	@pip install -r requirements-dev.txt
 
+
+
 docs: ## build HTML documentation
 	sphinx-build docs/ docs/_build/
 
+
+
 hook/%: WIKI_PATH?=../wiki
 hook/%: ## run post-build hooks for an image
 	python3 -m tagging.tag_image --short-image-name "$(notdir $@)" --owner "$(OWNER)" && \
 	python3 -m tagging.create_manifests --short-image-name "$(notdir $@)" --owner "$(OWNER)" --wiki-path "$(WIKI_PATH)"
-
 hook-all: $(foreach I,$(ALL_IMAGES),hook/$(I) ) ## run post-build hooks for all images
 
-img-clean: img-rm-dang img-rm ## clean dangling and jupyter images
 
+
+img-clean: img-rm-dang img-rm ## clean dangling and jupyter images
 img-list: ## list jupyter images
 	@echo "Listing $(OWNER) images ..."
 	docker images "$(OWNER)/*"
-
 img-rm: ## remove jupyter images
 	@echo "Removing $(OWNER) images ..."
 	-docker rmi --force $(shell docker images --quiet "$(OWNER)/*") 2> /dev/null
-
 img-rm-dang: ## remove dangling images (tagged None)
 	@echo "Removing dangling images ..."
 	-docker rmi --force $(shell docker images -f "dangling=true" -q) 2> /dev/null
 
+
+
 pre-commit-all: ## run pre-commit hook on all files
 	@pre-commit run --all-files || (printf "\n\n\n" && git --no-pager diff --color=always)
-
 pre-commit-install: ## set up the git hook scripts
 	@pre-commit --version
 	@pre-commit install
 
+
+
 pull/%: DARGS?=
 pull/%: ## pull a jupyter image
 	docker pull $(DARGS) $(OWNER)/$(notdir $@)
-
-pull-all: $(foreach I,$(ALL_IMAGES),pull/$(I) ) ## pull all images
+pull-all: $(foreach I,$(ALL_IMAGES),pull/$(I)) ## pull all images
 
 push/%: DARGS?=
 push/%: ## push all tags for a jupyter image
+	@echo "::group::Push $(OWNER)/$(notdir $@) (amd64)"
 	docker push --all-tags $(DARGS) $(OWNER)/$(notdir $@)
-
+	@echo "::endgroup::Push $(OWNER)/$(notdir $@) (amd64)"
 push-all: $(foreach I,$(ALL_IMAGES),push/$(I) ) ## push all tagged images
 
+push-multi/%: DARGS?=
+push-multi/%: ## push all tags for a jupyter image that support multiple architectures
+	@echo "::group::Push $(OWNER)/$(notdir $@) (amd64,arm64)"
+	docker buildx build $(DARGS) --rm --force-rm $($(subst -,_,$(notdir $@))_EXTRA_TAG_ARGS) -t $(OWNER)/$(notdir $@):latest ./$(notdir $@) --build-arg OWNER=$(OWNER) --platform "linux/amd64,linux/arm64"
+	@echo "::endgroup::Push $(OWNER)/$(notdir $@) (amd64,arm64)"
+push-all-multi: $(foreach I,$(MULTI_IMAGES),push-multi/$(I)) $(foreach I,$(AMD64_ONLY_IMAGES),push/$(I)) ## push all tagged images
```

Comment:
This is push-all

Reply:
Like for `build-all` I think it's important to separate this from `push-all` that doesn't use `docker buildx`. I think `push-all` shouldn't be used any more by this repo officially any more, but I think it can be useful still if you maintain your own image or similar using this infrastructure and only want to build/push system images instead of the dual-platform images.